.github/workflows/pull_request_swift_matrix.yml

```diff
@@ -0,0 +1,84 @@
+name: Pull Request
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        type: string
+        description: "The name of the workflow used for the concurrency group."
+        required: true
+      matrix_linux_command:
+        type: string
+        description: "The command of the current Swift version linux matrix job to execute."
+        required: true
+      matrix_linux_5_8_enabled:
+        type: string
+        description: "The command of the 5.8 Swift version linux matrix job to execute."
+      matrix_linux_5_8_command_override:
+        type: string
+        description: "The command of the 5.8 Swift version linux matrix job to execute."
+      matrix_linux_5_9_enabled:
+        type: string
+        description: "The command of the 5.9 Swift version linux matrix job to execute."
+      matrix_linux_5_9_command_override:
+        type: string
+        description: "The command of the 5.9 Swift version linux matrix job to execute."
+      matrix_linux_5_10_enabled:
+        type: string
+        description: "The command of the 5.10 Swift version linux matrix job to execute."
+      matrix_linux_5_10_command_override:
+        type: string
+        description: "The command of the 5.10 Swift version linux matrix job to execute."
+      matrix_linux_nightly_next_enabled:
+        type: string
+        description: "The command of the nightly next Swift version linux matrix job to execute."
+      matrix_linux_nightly_next_command_override:
+        type: string
+        description: "The command of the nightly next Swift version linux matrix job to execute."
+      matrix_linux_nightly_main_enabled:
+        type: string
+        description: "The command of the nightly main Swift version linux matrix job to execute."
+      matrix_linux_nightly_main_command_override:
+        type: string
+        description: "The command of the nightly main Swift version linux matrix job to execute."
+
+## We are cancelling previously triggered workflow runs
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ inputs.name }}
+  cancel-in-progress: true
+ 
+jobs:
+  linux:
+    name: ${{ matrix.swift.swift_version }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        swift:
+        - image: swift:5.8-jammy
+          swift_version: "5.8"
+        - image: swift:5.9-jammy
+          swift_version: "5.9"
+        - image: swift:5.10-jammy
+          swift_version: "5.10"
+        - image: swiftlang/swift:nightly-6.0-jammy
+          swift_version: "nightly-next"
```

Comment:
Should the version be more specific? e.g. "nightly-6.0"?

Reply:
The next version number is often decided quite close to the release so I wanted to keep it general with `next`

Reply:
As an example the next release was called swift `5.11` for a long time until it got renamed to swift 6.

Reply:
I just changed it to `nightly-6.0` now.

Reply:
But there was never a 5.11 branch or docker image, it only appeared in `swiftc --version`. At this point, where we're selecting a docker image we _must_ know the version (apart from `main` builds)